internal/controllers/clustertemplate_controller.go

```diff
@@ -239,6 +253,69 @@ func validateConfigmapReference[T any](
 	return nil
 }
 
+// validateName return true if the ClusterTemplate name is the
+// format: <name>.<version>, false otherwise
+func validateName(c client.Client, name, version, metadataName, namespace string) error {
+	if metadataName != name+"."+version {
```

Comment:
This would work well as an XValidation rule.

Reply:
Fixed